@here/harp-mapview/lib/BackgroundDataSource.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TileKey, TilingScheme, webMercatorTilingScheme } from "@here/harp-geoutils";
+import { DataSource } from "./DataSource";
+import { TileGeometryCreator } from "./geometry/TileGeometryCreator";
+import { MapView, MapViewEventNames } from "./MapView";
+import { Tile } from "./Tile";
+
+/**
+ * Provides background geometry for all tiles.
+ */
+export class BackgroundDataSource extends DataSource {
+    private m_themeLoaded: boolean = false;
+
+    constructor(name: string = "background") {
+        super(name, undefined, 1, 20);
+        this.cacheable = true;
+    }
+
+    attach(mapView: MapView): void {
+        super.attach(mapView);
+        mapView.addEventListener(MapViewEventNames.ThemeLoaded, () => {
+            this.m_themeLoaded = true;
+        });
+    }
+
+    getTilingScheme(): TilingScheme {
+        return webMercatorTilingScheme;
```

Comment:
Proposal to resolve this:
```
getTilingScheme() {
     const other =    this.mapView.enabledDataSources.find(ds => ds !== this);
     return other && other.tilingScheme || webMercator;
}
```

Reply:
makes sense, will do.

Reply:
@ninok , @atomicsulfate Agree, that care should be taken to maintain runtime stability. My proposal was just pseudo-code (should've mention this).

We should memoize if detected tiling scheme changed since previous frame and if, then `this.mapView.clearTileCache(this)`.

Reply:
@zbig @ninok The frustum intersection cache is no reused between frames, it's just used to avoid recalculating the intersection in the same frame multiple times.

Anyway, I'm working on a refactor now that will remove this cache since it's actually not needed. It's enough to order the data sources by tiling scheme, calculate the intersection, and reuse it.

Reply:
@atomicsulfate 
> The frustum intersection cache is no reused between frames

I already know this but we discuss other cache here.
If `BacgroundDataSource.getTilingScheme` would be implemented in naive way like in my proposition, it would be possible for tiling scheme to change from frame to frame, but tiles in tile cache are indexed by tiles morton code, so this cache would be invalid.

That's why "dynamic" tiling scheme in background data source requires that we detect tiling scheme change and invalidate tile cache.

> It's enough to order the data sources by tiling scheme, calculate the intersection, and reuse it.

 Good point +1

Reply:
done